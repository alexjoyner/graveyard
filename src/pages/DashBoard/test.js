import React from 'react';
import { shallow } from 'enzyme';
import { DashBoard } from '.';

describe('DashBoard Page', () => {
  let component;
  beforeEach(() => {
    component = shallow(<DashBoard />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  xit('Should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  it('Should have ISideBar behavior', () => {
    expect(component.find('Connect(TISideBar)')).toHaveLength(1);
  });
  it('SHould have a Header component', () => {
    expect(component.find('Header')).toHaveLength(1);
  });
  it('Should have a sidebar toggle button', () => {
    expect(component.find('Connect(TISideBarToggle)')).toHaveLength(1);
  });
  it('Should contain a test notification', () => {
    expect(component.find('TestNotification')).toHaveLength(1);
  });
  it('Should have a SideBar', () => {
    expect(component.find('Connect(TISideBar)').props().SideBar).toBeDefined();
  });
  it('Should have a page body', () => {
    expect(component.find('Connect(TISideBar)').props().Body).toBeDefined();
  });
});
