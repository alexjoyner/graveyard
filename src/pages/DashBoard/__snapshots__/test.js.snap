// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashBoard Page Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashBoard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <TestNotification />,
        <Header
          color="dark"
        >
          <Connect(TISideBarToggle)
            Button={[Function]}
          />
          <h1>
            OEE Master
          </h1>
        </Header>,
        <Connect(TISideBar)
          Body={[Function]}
          SideBar={[Function]}
        />,
        <Connect(BaseHistoricalGraphModal) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Connect(TISideBarToggle)
              Button={[Function]}
            />,
            <h1>
              OEE Master
            </h1>,
          ],
          "color": "dark",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "Button": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OEE Master",
            },
            "ref": null,
            "rendered": "OEE Master",
            "type": "h1",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "Body": [Function],
          "SideBar": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TestNotification />,
          <Header
            color="dark"
          >
            <Connect(TISideBarToggle)
              Button={[Function]}
            />
            <h1>
              OEE Master
            </h1>
          </Header>,
          <Connect(TISideBar)
            Body={[Function]}
            SideBar={[Function]}
          />,
          <Connect(BaseHistoricalGraphModal) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Connect(TISideBarToggle)
                Button={[Function]}
              />,
              <h1>
                OEE Master
              </h1>,
            ],
            "color": "dark",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "Button": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "OEE Master",
              },
              "ref": null,
              "rendered": "OEE Master",
              "type": "h1",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "Body": [Function],
            "SideBar": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
